### ISSUE TITLE
Rules: require a `,` after an introductory subordinate clause

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#411]
## Summary

Add a new rule (or extend an existing one) that flags sentences starting with an introductory subordinate clause when the required comma is missing, to enforce: "Missing comma after a subordinate clause?".

This is a common technical-writing issue and shows up frequently in documentation prose.

### Current (should be flagged)

```markdown
In case you do need to look something up we're taking our experience writing technical books and using it to make the best documentation we can.
```

### Expected

```markdown
In case you do need to look something up, we're taking our experience writing technical books and using it to make the best documentation we can.
```

## Expected behavior

- Report a diagnostic when a sentence begins with an introductory subordinate clause and the comma is missing before the main clause.
- Apply only to prose text (for example: paragraphs, blockquotes, list item text), and never to:
  - fenced code blocks
  - inline code spans
  - link destinations (URLs)

## Implementation notes

- Keep the detection conservative to reduce false positives.
- A practical first scope is to target common leading patterns (case-insensitive) like:
  - "In case ..."
  - "If ...", "When ...", "While ...", "Although ...", "Because ...", "Since ...", "After ...", "Before ...", "Unless ..."
- If an auto-fix is added, it should only trigger when the rule can confidently identify the boundary between the introductory clause and the main clause.

## Acceptance criteria

- The example above is flagged.
- No diagnostics are produced inside fenced code blocks or inline code.
- Rule is documented in the rules reference and can be configured/disabled like other rules.

## References

- https://github.com/rvben/rumdl/blob/main/docs/RULES.md
- https://github.com/rvben/rumdl/blob/main/docs/global-settings.md
- https://github.com/rvben/rumdl/blob/main/docs/inline-configuration.md

**Comment by rvben** [comment]
Closing as out of scope — see #408 for details.
